consult.el

```diff
@@ -932,6 +940,81 @@ Depending on the selected item OPEN-BUFFER, OPEN-FILE or OPEN-BOOKMARK will be u
   (interactive)
   (consult--buffer #'switch-to-buffer #'find-file #'bookmark-jump))
 
+;;;###autoload
+(defun consult-kmacro (arg)
+  "Run a chosen keyboard macro.  With prefix ARG, run the macro that many times.
+
+Macros containing mouse clicks can't be displayed properly.  To
+keep things simple, macros with an empty display string (e.g.,
+ones made entirely of mouse clicks) are not shown."
+  (interactive "p")
+  (if (or last-kbd-macro kmacro-ring)
+      (let* ((numbered-kmacros
+              (cl-loop
+               ;; The most recent macro is not on the ring, so it must be
+               ;;  explicitly included.
+               for kmacro in (cons (if (listp last-kbd-macro)
+                                       last-kbd-macro
+                                     (list last-kbd-macro
+                                           kmacro-counter
+                                           kmacro-counter-format))
+                                   kmacro-ring)
+               ;; 1. Format the macros.  Mouse clicks are removed
+               ;;    by the format function.
+               ;;
+               ;; 2. Give the macros with non-empty strings a number, which
+               ;;    we use for rotating the ring.
+               for index = 0 then (1+ index)
+               for formatted-kmacro
+               = (condition-case nil
+                     (format-kbd-macro (if (listp kmacro)
+                                           (cl-first kmacro)
+                                         kmacro)
+                                       1)
+                   ;; Recover from error from ‘edmacro-fix-menu-commands’.
+                   ;; In Emacs 27, it looks like mouse events are
+                   ;; silently skipped over.
+                   (error
```

Comment:
Can we just ignore such errors?

Reply:
The function `edmacro-fix-menu-commands`, which is called by `format-kbd-maro`, will signal the error "Macros with mouse clicks are not supported by this command" in Emacs 26.3. The signalling of this error will prevent the command `consult-kmacro` from completing, so `consult-kmacro` should respond to it.

I chose to just include the warning in the candidate list, so that the problematic macros could still be run with `consult-kmacro`. Other options include removing those candidates from the list.

Later versions of Emacs just skip over the mouse clicks, and so don't signal the error.

I don't know how often people use macros with mouse clicks, so I am not sure of the best way to respond to the error.

Reply:
Ok. I am not sure I understand everything you write but I think we should do the following: filter out all macros with mouse clicks in the beginning. And then don't catch any errors. Then we are fine. Nobody uses mouse clicks with macros and if someone does we could still add support later if the issue comes up. But I don't want to focus on special cases now. We should exclude it.